sdk/go/common/workspace/plugins.go

```diff
@@ -1017,7 +1017,12 @@ func NewPluginSpec(
 	}
 
 	urlRegex := regexp.MustCompile(`^[^\./].*\.[a-z]+/[a-zA-Z0-9-/]*[a-zA-Z0-9/]$`)
-	if urlRegex.MatchString(name) {
+	if strings.HasPrefix(name, "https://") || urlRegex.MatchString(name) {
+		// We support URLs with and without the https:// prefix.  Standardize them here, so we can work with
+		// them uniformly.
+		if strings.HasPrefix(name, "https://") {
+			name = strings.TrimPrefix(name, "https://")
+		}
```

Comment:
Would this also work as:

```go
// We support URLs with and without the https:// prefix.  Standardize them here, so we can work with
// them uniformly.
if strings.HasPrefix(name, "https://") {
	name = strings.TrimPrefix(name, "https://")
}
if urlRegex.MatchString(name) {
```

as in, just trim once at the top and leave the existing logic as-is?

Reply:
It would work in most cases, but it's not equivalent.  With the version I have you could e.g. say `pulumi package add https://localhost/some/repo.git`, and it would work, or `pulumi package add https://intranet/some/repo.git`. It's probably a very uncommon case, where there's no `.`s anywhere in the first part of the URL, so not sure if it's worth worrying about, but we might as well support it.  Now that I'm typing this out I think it would be good to add a testcase for this as well.

Reply:
That is a good catch! Worth a comment? I'm happy either way -- as you say the test case is the thing that matters. Nice! :boom:

Reply:
I don't think it is in this case.  It's a small detail that's best captured in tests, but doesn't deserve a long explanation in code imo.